### PR TITLE
New Website: Set up Google Analytics

### PR DESCRIPTION
**Pull request by andrew032011**
### Summary 
Largely copy paste from https://github.com/cornell-dti/idol/pull/283, adapted to work with `app/` directory instead of the older `pages/` directory model.

Configured a new stream on our google analytics account for new website and set the URL to be new.cornelldti.org while we're still pre-launch.

### Notion/Figma Link 
https://www.notion.so/Set-up-google-analytics-for-new-website-d8d425162cd24b29b48a248d948a8665?pvs=4

### Test Plan 
We can test this by hitting the website a bunch of times across a few days to see if any data is received. It can take up to 48 hours for data to begin showing.

### CODE REVIEW

File: new-dti-website/src/app/layout.tsx [hunk 1]
```diff
@@ -1,22 +1,58 @@
...
-      <div className="relative overflow-x-hidden">
-        <Page>{children}</Page>
-      </div>
-    </body>
-  </html>
-);
+const RootLayout = ({ children }: { children: React.ReactNode }): JSX.Element => {
+  const pathname = usePathname();
+
+  const handleRouteChange = (path: string) => {
```
Comment (andrew032011) [inline new-dti-website/src/app/layout.tsx:26]: This allows us to handle collecting usage data across different pages.

File: new-dti-website/src/app/layout.tsx [hunk 2]
```diff
@@ -1,22 +1,58 @@
...
+    window.gtag('config', MEASUREMENT_ID, {
+      page_path: path
+    });
+  };
+
+  useEffect(() => {
```
Comment (JasonMun7) [inline new-dti-website/src/app/layout.tsx:32]: This looks great to me Andrew! I was just wondering if were possible to make this into a hook maybe like 
` useGoogleAnalytics();` with `handleRouteChange` so the layout is more focused on rendering the components? I think this is definitely a future thing though! 

```suggestion
  useGoogleAnalytics(MEASUREMENT_ID);
```

**Review by JasonMun7 — APPROVED** [review]
This looks great to me Andrew! I just left a comment about something we can definitely look to in the future if we wanted!

File: new-dti-website/src/hooks/useGoogleAnalytics.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { useCallback, useEffect } from 'react';
```
Comment (JasonMun7) [inline new-dti-website/src/hooks/useGoogleAnalytics.ts:1]: This looks fantastic!

File: new-dti-website/src/consts.ts [hunk 4]
```diff
@@ -1,3 +1,4 @@
 export const MOBILE_BREAKPOINT = 640;
 export const TABLET_BREAKPOINT = 768;
 export const LAPTOP_BREAKPOINT = 1024;
+export const MEASUREMENT_ID = process.env.MEASUREMENT_ID ?? '';
```
Comment (JasonMun7) [inline new-dti-website/src/consts.ts:4]: Great idea placing this in the env!

**Review by JasonMun7 — APPROVED** [review]
This looks great to me!

**Pull request merged**